lib/ansible/modules/cloud/amazon/iam_managed_policy.py

```diff
@@ -289,6 +299,11 @@ def main():
     if not HAS_BOTO3:
         module.fail_json(msg='boto3 is required for this module')
 
+    if module.params.get('fail_on_delete'):
+        module.deprecate('The fail_on_delete option does nothing and'
```

Comment:
You don't need this anymore, a deprecation warning will automatically be issued when someone uses this option.